android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -2112,6 +2153,26 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
       }
       return false;
     }
+
+    private void handleGlobeKeyAction(boolean globeKeyDown) {
+      // Update globeKeyState
+      if (globeKeyState != GlobeKeyState.GLOBE_KEY_STATE_LONGPRESS) {
+        globeKeyState = globeKeyDown ? GlobeKeyState.GLOBE_KEY_STATE_DOWN : GlobeKeyState.GLOBE_KEY_STATE_UP;
+      }
+
+      if (KMManager.shouldAllowSetKeyboard()) {
```

Comment:
So is there any situation where we might have `shouldAllowSetKeyboard()` returning `false` and leaving us with a `globeKeyState` which is not `GLOBE_KEY_STATE_UP`?

Reply:
I don't think it's possible. But just in case I've added an else to clear `globeKeyState`